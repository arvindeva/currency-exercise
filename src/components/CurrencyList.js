import React from 'react';
import Currency from './Currency';

const CurrencyList = props => {

  return (
    <div>
      {props.currencyList.map(currency => (
        <Currency
          key={currency}
          currency={currency}
          exchangeRate={props.exchangeRates[currency]}
          usdInput={props.usdInput}
          onRemoveClick={props.onRemoveClick}
        />
      ))}
    </div>
  );
};

export default CurrencyList;
