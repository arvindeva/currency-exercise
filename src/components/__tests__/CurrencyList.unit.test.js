import React from 'react';
import { shallow } from 'enzyme';
import CurrencyList from '../CurrencyList';

describe('renders CurrencyList', () => {
  test('renders CurrencyList with empty list', () => {
    const currencyList = [];

    const wrapper = shallow(<CurrencyList currencyList={currencyList} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders CurrencyList with actual currencies and empty exchange rates', () => {
    const currencyList = ['USD', 'GBP', 'IDR'];
    const exchangeRates = {};
    const wrapper = shallow(
      <CurrencyList currencyList={currencyList} exchangeRates={exchangeRates} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('renders CurrencyList with actual currencies and actual exchange rates', () => {
    const currencyList = ['CAD', 'CHF', 'IDR'];
    const exchangeRates = {
      CAD: 1.3320535714,
      CHF: 0.9995535714,
      IDR: 14220
    };
    const wrapper = shallow(
      <CurrencyList currencyList={currencyList} exchangeRates={exchangeRates} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
