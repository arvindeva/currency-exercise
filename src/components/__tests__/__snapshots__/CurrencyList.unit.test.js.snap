// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CurrencyList renders CurrencyList with actual currencies and actual exchange rates 1`] = `
<div>
  <Currency
    currency="CAD"
    exchangeRate={1.3320535714}
    key="CAD"
  />
  <Currency
    currency="CHF"
    exchangeRate={0.9995535714}
    key="CHF"
  />
  <Currency
    currency="IDR"
    exchangeRate={14220}
    key="IDR"
  />
</div>
`;

exports[`renders CurrencyList renders CurrencyList with actual currencies and empty exchange rates 1`] = `
<div>
  <Currency
    currency="USD"
    key="USD"
  />
  <Currency
    currency="GBP"
    key="GBP"
  />
  <Currency
    currency="IDR"
    key="IDR"
  />
</div>
`;

exports[`renders CurrencyList renders CurrencyList with empty list 1`] = `<div />`;
